### ISSUE TITLE
[Synk: High] Arbitrary Code Execution found in numpy [due: 9/5/2019]

### ISSUE THREAD

**Issue by jason-upchurch** [fecgov/openFEC#3895]
## Summary ##

Found in `requirements-dev.txt`:

```
High severity vulnerability found in numpy
Description: Arbitrary Code Execution
Info: https://snyk.io/vuln/SNYK-PYTHON-NUMPY-73513
Introduced through: pandas@0.16.2
From: pandas@0.16.2 > numpy@1.16.2
```

## Overview ##
`numpy` is a fundamental package needed for scientific computing with Python.

Affected versions of this package are vulnerable to Arbitrary Code Execution. It uses the `pickle` Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a `numpy.load` call.

Remediation
Upgrade `numpy` to version 1.16.3 or higher.

**Comment by pkfec** [comment]
The vulnerability is present in NumPy versions between 1.10 (released in October of 2015) and the most recent version, 1.16. **_The arbitrary code execution vulnerability is fixed in numpy v0.16.3_**  by setting the flag `allow_pickle=False` .

**more info on vulnerability** : https://snyk.io/blog/numpy-arbitrary-code-execution-vulnerability/
**numpy v0.16.3 code changes** : https://github.com/charris/numpy/blame/8cea82a54a36b2232b484c7a50a32cefcb85e6bf/numpy/lib/npyio.py#L151

No need to make any changes to our current `pandas v0.16.2 `.  since pandas v0.16.2 comes with latest numpy v1.17.0.

**Comment by pkfec** [comment]
No changes required. Closing this issue.
